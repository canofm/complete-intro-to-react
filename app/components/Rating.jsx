// @flow

import React from 'react';
import Spinner from './Spinner';

const RatingValue = (props: { value: string }) => <h3>{props.value}</h3>;

const Rating = (props: { value?: string }) =>
  props.value ? <RatingValue {...props} /> : <Spinner />;

Rating.defaultProps = {
  value: ''
};

export default Rating;
