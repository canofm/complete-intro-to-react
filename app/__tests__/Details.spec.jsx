import React from 'react';
import { shallow } from 'enzyme';
import Details from '../components/Details';
import Rating from '../components/Rating';
import preload from '../../data.json';

describe('Details component', () => {
  it('should render correctly', () => {
    const component = shallow(<Details ratingComponent={Rating} show={preload.shows[0]} />);
    expect(component).toMatchSnapshot();
  });
});
