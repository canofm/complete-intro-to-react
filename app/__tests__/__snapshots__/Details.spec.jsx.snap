// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Details component should render correctly 1`] = `
<div
  className="details"
>
  <Header
    handleSearchTermChange={[Function]}
    searchTerm=""
    showSearch={false}
  />
  <section>
    <h1>
      Atlanta
    </h1>
    <h2>
      (
      2008â€“2013
      )
    </h2>
    <img
      alt="Poster for Atlanta"
      src="/public/img/posters/a.jpg"
    />
    <p>
      Two cousins, with different views on art versus commerce, on their way up through the Atlanta rap scene; "Earnest 'Earn' Marks," an ambitious college drop-out and his estranged cousin, who suddenly becomes a star.
    </p>
  </section>
  <div>
    <iframe
      allowFullScreen={true}
      frameBorder="0"
      src="https://www.youtube-nocookie.com/embed/MpEdJ-mmTlY?rel=0&amp;controls=0$amp;showinfo=0"
      title="Trailer for Atlanta"
    />
  </div>
</div>
`;
